c10/core/CPUAllocator.cpp

```diff
@@ -37,28 +37,66 @@ void memset_junk(void* data, size_t num) {
   }
 }
 
+void* alloc(size_t nbytes) {
+  if (nbytes == 0) {
+    return nullptr;
+  }
+
+  void* data;
+#ifdef __ANDROID__
+    data = memalign(gAlignment, nbytes);
+#elif defined(_MSC_VER)
+    data = _aligned_malloc(nbytes, gAlignment);
+#else
+    CAFFE_ENFORCE_EQ(posix_memalign(&data, gAlignment, nbytes), 0);
+#endif
+
+  CAFFE_ENFORCE(
+      data,
+      "DefaultCPUAllocator: not enough memory: you tried to allocate %dGB. Buy new RAM!",
+      nbytes / 1073741824);
+  // move data to a thread's NUMA node
+  NUMAMove(data, nbytes, GetCurrentNUMANode());
+  CHECK(
+      !FLAGS_caffe2_cpu_allocator_do_zero_fill ||
+      !FLAGS_caffe2_cpu_allocator_do_junk_fill)
+    << "Cannot request both zero-fill and junk-fill at the same time";
+  // Only fill data if we are running fbcode caffe2 code
+#ifdef C10_FBCODE_CAFFE2
```

Comment:
If you are going to ifdef this, you should ifdef the check above, and you should also ifdef away the flags so that they're not available in OSS.